### PR TITLE
Add fish support

### PR DESCRIPTION
**Pull request by superm1**
This reverts some of 8ea478a ("fish: Remove fish support") but fixes the unit tests and adds coverage for all remaining commands

Fixes: https://github.com/kworkflow/kworkflow/issues/1165

### CODE REVIEW

File: tests/unit/fish_completion_test.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+. ./tests/unit/utils.sh --source-only
```
Comment (rodrigosiqueira) [inline tests/unit/fish_completion_test.sh:3]: We have a helper for that, you can just use:

`include './tests/unit/utils.sh'`

**Comment by superm1** [comment]
(Will close this and open a new one targeting correct branch)

File: tests/unit/fish_completion_test.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+function suite()
+{
+  suite_addTest "testKwCompletion"
```
Comment (rodrigosiqueira) [inline tests/unit/fish_completion_test.sh:7]: Maybe use a more meaningful name for this test, for example: `testKwCompletionForFish`.

P.s.: I never used this suite_addTest, do we really need it here?

File: tests/unit/fish_completion_test.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+function getSortedCompletions()
```
Comment (rodrigosiqueira) [inline tests/unit/fish_completion_test.sh:10]: We use snake case in kw.

**Review by rodrigosiqueira — COMMENTED** [review]
I just publish my review because I noticed that you close the PR when I was reviewing it.

**Comment by superm1** [comment]
> I just publish my review because I noticed that you close the PR when I was reviewing it.

Oh sorry!

I figured I should target right branch before you got to it.  I'll adjust the feedback for the new branch too.

**Comment by rodrigosiqueira** [comment]
> > I just publish my review because I noticed that you close the PR when I was reviewing it.
> 
> Oh sorry!
> 
> I figured I should target right branch before you got to it. I'll adjust the feedback for the new branch too.

np, I'll review the new PR after you update it.

Thanks

**Pull request closed without merging**